mlapi/question_generator.py

```diff
@@ -0,0 +1,49 @@
+from collections import Counter
+from mlapi.model.question import Question
+
+
+class QuestionGenerator(object):
+    def generate_questions(self, facets_by_document):
+        if len(facets_by_document) is 0:
+            return []
+
+        values_by_name = self.get_discriminating_facets(facets_by_document)
+        questions = []
+        id = 0
+        for key, values in values_by_name.items():
+            questions.append(Question(id=id, facet_name=key, facet_values=values, answer="", status=""))
+            id += 1
+        return questions
+
+    def get_discriminating_facets(self, facets_by_document):
+        unique_facets_by_document = self.remove_redundancies_in_documents(facets_by_document)
+        facets = []
```

Comment:
probably equivalent to simpler:
facets = unique_facets_by_document.values()

Reply:
this one won't work facets becomes a list of list